projects/pgai/pgai/alembic/autogenerate.py

```diff
@@ -0,0 +1,341 @@
+    if hasattr(metadata, "info"):
+        vectorizers = metadata.info.get("vectorizers", {})
+        for _key, config in vectorizers.items():
+            source_schema = config.get("source_schema") or "public"
+            target_table = config["target_table"]
+            if "." not in target_table:
+                target_table = f"{source_schema}.{target_table}"
+            model_vectorizers[target_table] = config
+
+    # Compare and generate operations
+    for table_name, model_config in model_vectorizers.items():
+        if table_name not in existing_vectorizers:
+            # Create new vectorizer
+            upgrade_ops.ops.append(
+                _create_vectorizer_op(model_config)
+            )
+        else:
+            # Check for configuration changes
+            existing_config = existing_vectorizers[table_name]
+            if _config_has_changed(model_config, existing_config):
+                # Drop and recreate vectorizer if config changed
+                upgrade_ops.ops.extend(
+                    [
+                        DropVectorizerOp(
+                            existing_config["id"],
+                            drop_all=True,
+                        ),
+                        _create_vectorizer_op(model_config),
+                    ]
+                )
+
+    for table_name, existing_config in existing_vectorizers.items():
+        if table_name not in model_vectorizers:
+            upgrade_ops.ops.append(
+                DropVectorizerOp(
+                    existing_config["id"],
+                    drop_all=True,
+                )
+            )
+
+
+def _create_vectorizer_op(
+        model_config: dict[str, Any]
+) -> CreateVectorizerOp:
+    return CreateVectorizerOp(
+        source_table=model_config["source_table"],
+        destination=model_config.get("destination"),
+        embedding=EmbeddingConfig(**model_config["embedding"]),
+        chunking=ChunkingConfig(**model_config["chunking"]),
+        formatting_template=model_config.get(
+            "formatting_template", "$chunk"
+        ),
+        indexing=DiskANNIndexingConfig(**model_config.get("indexing", {}))
+        if "indexing" in model_config
+        else None,
+        scheduling=SchedulingConfig(**model_config.get("scheduling", {}))
+        if "scheduling" in model_config
+        else None,
+        processing=ProcessingConfig(**model_config.get("processing", {}))
+        if "processing" in model_config
+        else None,
+        target_schema=model_config.get("target_schema"),
+        target_table=model_config.get("target_table"),
+        view_schema=model_config.get("view_schema"),
+        view_name=model_config.get("view_name"),
+        queue_schema=model_config.get("queue_schema"),
+        queue_table=model_config.get("queue_table"),
+        grant_to=model_config.get("grant_to"),
+        enqueue_existing=model_config.get("enqueue_existing", True),
+    )
+    
+
+def _config_has_changed(
```

Comment:
I wonder if this can be as easy as rendering the two configs as create_vectorizer calls and doing a string diff.